src/components/TimeEntries.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, useState } from "react";
+
+import { deleteTimeEntries } from "../services/time-entries/deleteTimeEntries";
+
+import * as Styled from "./TimeEntries.styled";
+
+import * as Type from "../types/types";
+
+import { useStore } from "../stores/ZustandStore";
+import TimeEntry from "./TimeEntry";
+import DateRegistry from "./DateRegistry";
+
+const convertDates = (timeEntry: Type.TimeEntry) => {
+  const startDateTime = new Date(timeEntry.startTime);
+  const endDateTime = new Date(timeEntry.endTime);
+  const startTime = startDateTime.toLocaleTimeString("nl-NL", {
+    hour: "2-digit",
+    minute: "2-digit",
+  });
+  const endTime = endDateTime.toLocaleTimeString("nl-NL", {
+    hour: "2-digit",
+    minute: "2-digit",
+  });
+  return { timeRegistration: `${startTime} - ${endTime}`, startDateTime, endDateTime };
+};
+
+const calculateDuration = (timeEntry: Type.TimeEntry) => {
+  const startDateTime = new Date(timeEntry.startTime);
+  const endDateTime = new Date(timeEntry.endTime);
+  const durationAsNumber = endDateTime.getTime() - startDateTime.getTime();
+  return { durationAsNumber, duration: new Date(durationAsNumber).toISOString().substr(11, 5) };
+};
+
+function TimeEntries() {
+  const rawTimeEntries = useStore((state) => state.rawTimeEntries);
+  const setRawTimeEntries = useStore((state) => state.setRawTimeEntries);
+  const [timeEntries, setTimeEntries] = useState<Type.TimeEntry[]>([]);
+
+  async function deleteTimeEntry(_id: number) {
+    const newTimeEntries = rawTimeEntries.filter((entry) => entry.id !== _id);
+    setRawTimeEntries(newTimeEntries);
+    await deleteTimeEntries(_id);
+  }
+
+  useEffect(() => {
+    const newTimeEntries = [];
+    rawTimeEntries.map((timeEntry) =>
+      newTimeEntries.push({
```

Comment:
Try to avoid `.push()`.

Reply:
how should I change this? make newTimeEntries not a constant and use a spread like this:
[...newTimeEntries, timeEntry]